src/Symfony/Component/Validator/Constraints/LengthValidator.php

```diff
@@ -39,8 +39,14 @@ public function validate($value, Constraint $constraint)
 
         $stringValue = (string) $value;
 
-        if (!$invalidCharset = !@mb_check_encoding($stringValue, $constraint->charset)) {
-            $length = mb_strlen($stringValue, $constraint->charset);
+        try {
+            $invalidCharset = !@mb_check_encoding($stringValue, $constraint->charset);
+        } catch (\ValueError $e) {
+            if (!str_starts_with($e->getMessage(), 'mb_check_encoding(): Argument #2 ($encoding) must be a valid encoding')) {
```

Comment:
Shouldn't we throw when this message matches? Otherwise the user will see a constraint violation for a case where they cannot do anything about (a misconfigured charset).

Reply:
or a bad input charset, isn't it?

I don't think we should change the behavior that is in place currently.
This patch just makes it work the same under PHP 8.

Of course, we can consider changing the behavior in case of a bad input charset, but we should do it consistently for all versions of PHP if we do.

Reply:
@xabbuh In the PR description, I wrote:

> Maybe we can reconsider this behavior for Symfony 5.2. It feels a bit odd to me because we swallow a potential misconfiguration of the validator.

I didn't want to break existing behavior on an old LTS branch.

Reply:
okay, makes sense